src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/Tolerance/index.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect } from 'react';
+import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
+import { Alert, Form } from '@edx/paragon';
+import PropTypes from 'prop-types';
+import SettingsOption from '../../SettingsOption';
+import messages from './messages';
+import { ToleranceTypes } from './constants';
+
+// eslint-disable-next-line no-unused-vars
+export const isAnswerRangeSet = ({ answers }) =>
+  // TODO: for TNL 10258
+  // eslint-disable-next-line implicit-arrow-linebreak
+  false;
+
+export const handleToleranceTypeChange = ({ updateSettings, tolerance, answers }) => (event) => {
+  if (!isAnswerRangeSet(answers)) {
+    let value;
+    if (event.target.value === ToleranceTypes.none.type) {
+      value = null;
+    } else {
+      value = tolerance.value || 0;
+    }
+    const newTolerance = { type: event.target.value, value };
+    updateSettings({ tolerance: newTolerance });
+  }
+};
+
+export const handleToleranceValueChange = ({ updateSettings, tolerance, answers }) => (event) => {
+  if (!isAnswerRangeSet(answers)) {
+    const newTolerance = { value: event.target.value, type: tolerance.type };
+    updateSettings({ tolerance: newTolerance });
+  }
+};
+
+export const getSummary = ({ tolerance, intl }) => {
+  switch (tolerance?.type) {
+    case ToleranceTypes.percent.type:
+      return `± ${tolerance.value}%`;
```

Comment:
This renders as expected on the load. But when I switch from another tolerance type to percent, the summary reads "None". It looks like `tolerance.type` equals "Percentage" instead of "Percent".